### ISSUE TITLE
Please continue work on Aero Glass and make it a separate project

### ISSUE THREAD

**Issue by TheDarkBomber** [valinet/WinCenterTitle#11]
Your work on the glass branch of this project specifically is amazing so far, and I can't wait to see it when it has been fully developed. However, your last commit to it has been in November 2020, and I am worried that you have given up working on it, which would be a shame, as it is the only way to get Aero Glass on Windows 10 2004+ discounting Stardock which is paywalled. Also, a few small bug reports: DWM opens as a console window, closing it just restarts DWM and the console window, and title text has an opaque white background on most windows. Asides from that, there are only minor issues which are due to it not being fully developed yet. You should also really consider making the glass branch into its own project, which would make it more visible when people search for « Aero Glass for Windows 10 » on search engines. I only found about this by searching for something which would give me centered title text (which it does excellently), and then seeing the issue where someone was asking for left-aligned text, which showed me the Aero Glass they had which led me to find out about the glass branch. Please make this is a separate project so more people know about this, please… and please continue work on it, it would be very much appreciated.

**Comment by valinet** [comment]
Thanks for the kind words.

Indeed, visibility for this is kind of low, but the thing is the project is, as you said, in an unfinished state. The console window can easily be taken care of, just comment the ‘AllocConsole’ line (the console is there for debugging purposes). Others, like the white background color for the title bar text, or the oversized borders, require a ton more research work to take care of. 

The problem is, I don’t have much time to work on this at the moment, unfortunately. No to mention it is pretty tedious and kind of unrewarding, as it doesn’t really teach me much yet it takes a lot of time and effort to do - basically, one has to study the disassembly of dwm augmented with some ‘hints’, but it is nowhere near source code level, and even having access to source code it would be pretty hard to understand such a complex program, and that in a timely manner. So, to summarize, it takes lots of time and effort, and I don’t have mainly that much time at the moment, being very busy at work, not to mention I kind of lost the interest in working on it. It was more like an experiment, and at the time I was more free regarding stuff to do at my workplace. Furthermore, maintaining this would be a nightmare: basically, it would require manual patching and even rewrites at pretty much every new Windows release - Microsoft actually updates dwm, that’s why BM’s Aero Glass broke in 2004. Even if he had released the source code, people would have had a hard time making it to work in 2004, as things changed quite a bit regarding the pipeline he was hooking into, for example. Moreover, even the main WinCenterTitle I expect to brake on Windows vNext, but hopefully that’s easier to fix as it messes less with dwm. The thing is pretty much documented only in my head, like, I don’t even remember what I have done exactly in the glass version anymore. Writing good docs, again, which this requires, takes a lot of time and effort.

So yeah, again, this would require a lot of effort, which, to be frank, I don’t have time to do myself alone. I posted this on GitHub in the hope that some people could actually contribute and we would split the work and this would become a community project. That’s the only way this can become viable, in my opinion. But this hasn’t happened yet.

So, unfortunately, I can’t do much at the moment. My schedule is full at work at the moment. And I can’t quit my job, even if, hypothetically, I made enough money from donations working on Aero Glass, I wouldn’t quit my job as I like it :)). Also, as much as I myself would like it, the “reward” I’d get compared to the amount of work, which is not even innovative, would make the victory bittersweet. And I always find side projects to work on, as there are always things I find no app or utility for, yet my workflow requires. Just to give you an example: I have 2 identical monitors with built in speakers and I want to output sound to both of them at the same time. I tried the apps that pop up on Google search, they don’t work on newer Windows 10 builds, so what’s there to do? Write my own stuff. And there’s more, only if I had more time and maybe energy. No matter how much less you sleep, the day still has got only 24 hours.

**Comment by TheDarkBomber** [comment]
Yes, I think that people with knowledge of DWM and the Windows API should contribute to this project in significant ways. Personally, I do not need to use Windows on versions above 2004, since it supports all the features I need, such as Hyper-V compatibility with VMware and WSL2. I also do not mind the other quirks like the oversized borders, only the title text background being opaque is a problem. Anyway, I wish you luck in getting other developers to assist you with this project.
